### PR TITLE
test: protect from environment leaking between tests

### PR DESCRIPTION
**Pull request by Rauno56**
# Description

Implemented an utility to clear relevant environment variables before running the tests.
Current ones are fortunately well-implemented and didn't have any errors regarding that, but
I stumbled upon fragile tests adding new in another PR.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Internal change (a change which is not visible to the package consumers)
- [ ] Documentation change or requires a documentation update

# How Has This Been Tested?

- [ ] Tested manually
- [x] Added automated tests

# Checklist:

- [x] Unit tests have been added/updated
- [ ] Documentation has been updated
- [ ] Change file has been generated (`npm run change:new`)

### CODE REVIEW

File: test/options.test.ts [hunk 1]
```diff
@@ -37,8 +37,11 @@ import {
   Options,
   defaultPropagatorFactory,
 } from '../src/options';
+import * as utils from './utils';
 
 describe('options', () => {
+  beforeEach(utils.cleanEnvironment);
```
Comment (jtmal-signalfx) [inline test/options.test.ts:43]: let's clean after tests, not before, this suggests that we don't know where the "messy" environment settings come from

File: test/resource.test.ts [hunk 2]
```diff
@@ -19,19 +19,11 @@ import * as sinon from 'sinon';
...
   beforeEach(() => {
-    process.env.OTEL_RESOURCE_ATTRIBUTES = '';
-  });
-
-  afterEach(() => {
-    if (envStub) {
-      envStub.restore();
-    }
-    delete process.env.OTEL_RESOURCE_ATTRIBUTES;
+    utils.cleanEnvironment();
```
Comment (jtmal-signalfx) [inline test/resource.test.ts:26]: again, let's clean after, this applies everywhere in the PR

**Review by jtmal-signalfx — APPROVED** [review]

File: test/options.test.ts (hunk 1 above)
Comment (Rauno56) [reply test/options.test.ts:43]: > suggests that we don't know where the "messy" environment settings come from
Well... we don't. And I feel bad for it honestly and would focus on making sure our tests run in a clean environment.

The environment is controlled by the runner. We don't assure the order of the tests executed so any of our tests might be the first, if your personal environment contains `OTEL_` prefixed stuff it already is affecting our tests - we should make sure we avoid falling into that.

Then... taking into account that we have to do before either way I don't see a reason to do it after as well.

Comment (jtmal-signalfx) [reply test/options.test.ts:43]: Yeah if you think that's the right way to go then merge.

**Pull request merged**